Sources/VLCEditController.swift

```diff
@@ -86,33 +89,23 @@ private extension VLCEditController {
 // MARK: - VLCEditToolbarDelegate
 
 extension VLCEditController: VLCEditToolbarDelegate {
-    func createPlaylist() {
-        if let model = model as? PlaylistModel {
-            let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
-                placeHolder: "NEW_PLAYLIST")
-
-            presentTextFieldAlert(with: alertInfo, completionHandler: {
-                text -> Void in
-                    model.create(name: text)
-                })
-
-        } else if let model = model as? VideoModel {
-            let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
-                                               placeHolder: "NEW_PLAYLIST")
-
-            presentTextFieldAlert(with: alertInfo, completionHandler: {
-                [selectedCellIndexPaths, model] text -> Void in
-                let playlist = model.medialibrary.createPlaylist(with: text)
-                for indexPath in selectedCellIndexPaths {
-                    if let media = model.anyfiles[indexPath.row] as? VLCMLMedia {
-                        playlist.appendMedia(withIdentifier: media.identifier())
-                    }
+    func addToPlaylist(_ edittoolbar: VLCEditToolbar) {
+        let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
+                                           alertDescription: NSLocalizedString("PLAYLIST_DESCRIPTION", comment: ""),
+                                           placeHolder: NSLocalizedString("PLAYLIST_PLACEHOLDER", comment:""))
+
+        presentTextFieldAlert(with: alertInfo, completionHandler: {
+            [selectedCellIndexPaths, model] text -> Void in
+            let playlist = model.medialibrary.createPlaylist(with: text)
+            for indexPath in selectedCellIndexPaths {
+                if let media = model.anyfiles[indexPath.row] as? VLCMLMedia {
+                    playlist.appendMedia(withIdentifier: media.identifier())
```

Comment:
I realized that this removes the ability to create an empty playlist.

Do we want to remove this?

Reply:
Yes for now we want to remove it BUT I have thought about this because we lack not only the ability to create an empty playlist but also the ability to add to existing ones. hence I wrote : https://code.videolan.org/videolan/vlc-ios/issues/411

Reply:
hm thinking about it, having to present a viewcontroller makes another good case for making the Edittoolbarcontroller a UIViewcontroller and a childviewController of the MediaCategoryViewController

Reply:
> we lack not only the ability to create an empty playlist ...

Just to be sure, by "we lack" do you mean UI-wise or medialibrary-wise? 🤔